src/status_im/ui/screens/intro/views.cljs

```diff
@@ -294,31 +320,147 @@
+        (i18n/label-pluralize passphrase-word-count :t/words-n)]
+       (when-not next-button-disabled?
+         [react/view {:style {:background-color colors/green-transparent-10
+                              :border-radius 12
+                              :width 24
+                              :justify-content :center
+                              :align-items :center
+                              :height 24}}
+          [vector-icons/tiny-icon :tiny-icons/tiny-check {:color colors/green}]])])
+    [react/text {:style {:color      colors/gray
+                         :font-size  14
+                         :margin-bottom 8
+                         :text-align :center}}
+     (i18n/label :t/multiaccounts-recover-enter-phrase-text)]]
+   (when processing?
+     [react/view {:flex 1 :align-items :center}
+      [react/activity-indicator {:size      :large
+                                 :animating true}]
+      [react/text {:style {:color      colors/gray
+                           :margin-top 8}}
+       (i18n/label :t/processing)]])])
+
+(defn recovery-success [wizard-state]
+  (let [pubkey (get-in wizard-state [:derived constants/path-whisper-keyword :publicKey])]
+    [react/view {:flex           1
+                 :justify-content  :space-between
+                 :background-color colors/white}
+     [react/view {:flex            1
+                  :justify-content :space-between
+                  :align-items     :center}
+      [react/view {:flex-direction  :column
+                   :flex            1
+                   :justify-content :center
+                   :align-items     :center}
+       [react/view {:margin-horizontal 16
+                    :flex-direction    :column}
+        [react/view {:justify-content :center
+                     :align-items     :center
+                     :margin-bottom   11}
+         [react/image {:source {:uri (identicon/identicon pubkey)}
+                       :style  {:width         61
+                                :height        61
+                                :border-radius 30
+                                :border-width  1
+                                :border-color  colors/black-transparent}}]]
+        [react/text {:style           {:text-align  :center
+                                       :color       colors/black
+                                       :font-weight "500"}
+                     :number-of-lines 1
+                     :ellipsize-mode  :middle}
+         (gfy/generate-gfy pubkey)]
+        [react/text {:style           {:text-align  :center
+                                       :margin-top  4
+                                       :color       colors/gray
+                                       :font-family "monospace"}
+                     :number-of-lines 1
+                     :ellipsize-mode  :middle}
+         (utils/get-shortened-address pubkey)]]]]]))
+
+(defn intro-wizard [{:keys [step generating-keys?
```

Comment:
I get the idea but this needs to be a form-2 component and the step needs to be outside of the renderer, so that it doesn't change based on navigation actions, because that is the role of react-navigation

Reply:
I've reverted back to the idea of separate views for each wizard step, just reusing common top-bar/bottom-bar components. And having minimal subscriptions for each page. This was the only solution i've found to the perf problem. Otherwise we're going to have multiple wizard views being updated when e.g. navigating back (as they're all in navigation stack)

Reply:
Pushed current version, will polish it a bit more.